web/src/components/organisms/Project/settings/hooks.ts

```diff
@@ -62,9 +63,11 @@ export default ({ projectId }: Params) => {
         Notification.error({ message: t("Failed to update project.") });
         return;
       }
+      const updatedProjectData = project.data.updateProject.project;
+      setCurrentProject({ ...updatedProjectData });
```

Comment:
I was trying to handle setting currentProject in the header, and only the header. 
Can you add a refetch to the updateProject mutation here and then take a look there to see if you can add/edit a dependency in the useEffect in the header that would update the currentProject after the cache is updated?

Reply:
@KaWaite , these lines:
https://github.com/reearth/reearth-cms/blob/273a9e513780a133e01099a01050682b3b3a3479/web/src/components/organisms/CMSWrapper/hooks.ts#L94
will prevent refetch (because the `projectId` is equal to `currentProject?.id` after the update) 
and:
https://github.com/reearth/reearth-cms/blob/273a9e513780a133e01099a01050682b3b3a3479/web/src/components/organisms/CMSWrapper/hooks.ts#L99

Will prevent updating the state for same reason, any suggestions, and what the reason for these lines?

Reply:
Can you actually remove those checks that are checking if projectId equals/doesn't equal currentProject, then do as I suggested above with the refetch?
I added those checks so that there weren't unneeded refetches or updating the currentProject, but I realize they are maybe unnecessary/not good.

Reply:
I tried to run the code without it, doesn't seem to affect.
I have updated the code for those parts.

Reply:
Is your linter working correctly? Since you removed the currentProject?.id check from the useEffect, you should be getting warnings to remove the dependency since its not referenced any more in the useEffect

Reply:
I'll check and get back